src/WaveTrack.cpp

```diff
@@ -1408,7 +1412,11 @@ void WaveTrack::Paste(double t0, const Track *src)
             newClip->Resample(mRate);
             newClip->Offset(t0);
             newClip->MarkChanged();
-            newClip->SetName(MakeClipCopyName(clip->GetName()));
+            if (pastingFromClipboard)
+               newClip->SetName(MakeClipCopyName(clip->GetName()));
+            else if (pastingFromTempTrack)
+               newClip->SetName(MakeNewClipName());
+            //else { leave clip name as is, track orginates from another track list }
```

Comment:
These changes go beyond fixing the referenced issue.  If there was agreement about new naming, I don’t know where it is.

Reply:
No, the initial design document didn't consider generated clips at all.